Domain.Sql.Tests/ScheduleCommandsDiagnosticTests.cs

```diff
@@ -0,0 +1,191 @@
+        }
+
+        private const string defaultClockName = "default";
+        private SqlEventSourcedRepository<CustomerAccount> accountRepository;
+        private SqlEventSourcedRepository<Order> orderRepository;
+        private FakeEventBus bus;
+        private CompositeDisposable disposables;
+        private SqlCommandScheduler scheduler;
+
+        [TestFixtureSetUp]
+        public void TestFixtureSetUp()
+        {
+            Logging.Configure();
+            Formatter<SchedulerAdvancedResult>.RegisterForAllMembers();
+            Formatter<CommandSucceeded>.RegisterForAllMembers();
+            Formatter<CommandFailed>.RegisterForAllMembers();
+
+            // disable authorization checks
+            Command<CustomerAccount>.AuthorizeDefault = (order, command) => true;
+            Command<Order>.AuthorizeDefault = (order, command) => true;
+        }
+
+        private void TriggerExceptionOnOrderCommands()
+        {
+            orderRepository.GetEventStoreContext = () => { throw new Exception(); };
+        }
+
+        private void StopTriggeringExceptions()
+        {
+            orderRepository.GetEventStoreContext = () => new EventStoreDbContext();
+        }
+
+        private void ClearAbandonedAndPendingScheduledCommands()
+        {
+            Debug.WriteLine("Clearing�abandoned�and�pending�scheduled�commands.");
+            using (var db = new CommandSchedulerDbContext())
+            {
+                var commands = db.ScheduledCommands.Where(s => s.AppliedTime == null).ToList();
+                commands.ForEach(
+                                 s =>
+                                 {
+                                     s.AppliedTime = s.DueTime ?? Clock.Now();
+                                     s.FinalAttemptTime = null;
+                                 });
+
+                db.SaveChanges();
+            }
+        }
+
+        private async Task TriggerAbandonedCommandByException()
+        {
+            var order = CommandSchedulingTests.CreateOrder();
+            await accountRepository.Save(new CustomerAccount(order.CustomerId).Apply(new ChangeEmailAddress(Any.Email())));
+            await orderRepository.Save(order);
+            TriggerExceptionOnOrderCommands();
+
+            var orderScheduler = Configuration.Current.CommandScheduler<Order>();
+            await orderScheduler
+                .Schedule(new CommandScheduled<Order>
+                          {
+                              AggregateId = order.Id,
+                              Command = new Cancel()
+                          });
+
+            for (var i = 1; i < 10; i++)
+            {
+                await AdvanceDefaultClock(TimeSpan.FromHours(1));
+            }
+
+            StopTriggeringExceptions();
+        }
+
+        private async Task TriggerOverdueCommandByException()
+        {
+            StopTriggeringExceptions();
+            var order = CommandSchedulingTests.CreateOrder();
```

Comment:
We could probably extract line 129 to 140 to a separate method since it's used in TriggerAbandonedCommandByException too